### PR TITLE
Add Support for GitHub enterprise as a build definition repository type

### PR DESCRIPTION
**Pull request by shubydo**
## All Submissions:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.
* [x] All new and existing tests passed.
* [x] My code follows the code style of this project.
* [x] I ran lint checks locally prior to submission.
* [x] Have you checked to ensure there aren't other open PRs for the same update/change?

## What about the current behavior has changed?

Add Support for GitHub enterprise as a build definition repository type.

Under the `repository` block in the `azuredevops_build_definition` resource, you can now specify `GitHubEnterprise` as the `repo_type` and specify the Github Enterprise URL using the `github_enterprise_url` .

Example:
```hcl
resource "azuredevops_build_definition" "sample_dotnetcore_app_release" {
  project_id = azuredevops_project.project.id
  name       = "Sample Build Definition"
  path       = "\\ExampleFolder"

  ci_trigger {
    use_yaml = true
  }

  repository {
    repo_type             = "GitHubEnterprise"
    repo_id               = "<GitHub Org>/<Repo Name>"
    github_enterprise_url = "https://github.company.com"
    branch_name           = "master"
    yml_path              = "azure-pipelines.yml"
    service_connection_id = "..."
  }

}
```

Issue Number: #35 

## Does this introduce a change to `go.mod`, `go.sum` or `vendor/`?

- [ ] Yes
- [x] No

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Any relevant logs, error output, etc?

(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

### CODE REVIEW

**Comment by xuzhang3** [comment]
Hi @Shubydo  , I'm afraid this new feature might not working if `repo_type` is `GitHubEnterprise` because ADO provider doesn't support create GitHub enterprise service connection yet.  
During build definition creation, service will use the service connection connect to GitHub enterprise and fetch the repository meta data. If service connection doesn't exist,   the creation will fail. One possible way is  using an existed project and GitHub enterprise service connection.

**Comment by shubydo** [comment]
> Hi @Shubydo , I'm afraid this new feature might not working if `repo_type` is `GitHubEnterprise` because ADO provider doesn't support create GitHub enterprise service connection yet.
> During build definition creation, service will use the service connection connect to GitHub enterprise and fetch the repository meta data. If service connection doesn't exist, the creation will fail. One possible way is using an existed project and GitHub enterprise service connection.

@xuzhang3 Yes, currently the ADO provider doesn't support creating GitHub enterprise service connections, but I was able to use the ID of an existing GitHub enterprise service connection (created through the console/UI) when creating the build definition. 

Also, there is a check to validate that a service connection ID is used when `repo_type` is `GitHubEnterprise` [here](https://github.com/terraform-providers/terraform-provider-azuredevops/pull/97/files#diff-2ca60e169df2b70d60401d8be8bc6e15R935-R937 ).

File: azuredevops/internal/service/build/resource_build_definition.go [hunk 1]
```diff
@@ -831,6 +848,12 @@ func expandBuildDefinition(d *schema.ResourceData) (*build.BuildDefinition, stri
 		repoURL = fmt.Sprintf("https://bitbucket.org/%s.git", repoID)
 		repoAPIURL = fmt.Sprintf("https://api.bitbucket.org/2.0/repositories/%s", repoID)
 	}
+	if strings.EqualFold(string(repoType), string(model.RepoTypeValues.GitHubEnterprise)) {
+		githubEnterpriseURL := repository["github_enterprise_url"].(string)
+		fmt.Println("githubEnterpriseURL: ", githubEnterpriseURL)
```
Comment (xuzhang3) [inline azuredevops/internal/service/build/resource_build_definition.go]: `fmt.Println` should removed or replace with `log.Printf`

File: website/docs/r/build_definition.html.markdown [hunk 2]
```diff
@@ -11,6 +11,7 @@ Manages a Build Definition within Azure DevOps.
 
 ## Example Usage
 
+### Tfs
```
Comment (xuzhang3) [inline website/docs/r/build_definition.html.markdown:14]: what does `Tfs` mean ?

File: website/docs/r/build_definition.html.markdown [hunk 3]
```diff
@@ -98,9 +122,10 @@ The following arguments are supported:
 
 - `branch_name` - (Optional) The branch name for which builds are triggered. Defaults to `master`.
 - `repo_id` - (Required) The id of the repository. For `TfsGit` repos, this is simply the ID of the repository. For `Github` repos, this will take the form of `<GitHub Org>/<Repo Name>`. For `Bitbucket` repos, this will take the form of `<Workspace ID>/<Repo Name>`.
-- `repo_type` - (Optional) The repository type. Valid values: `GitHub` or `TfsGit` or `Bitbucket`. Defaults to `Github`.
-- `service_connection_id` - (Optional) The service connection ID. Used if the `repo_type` is `GitHub`.
+- `repo_type` - (Optional) The repository type. Valid values: `GitHub` or `TfsGit` or `Bitbucket` or `GitHub Enterprise`. Defaults to `Github`.
```
Comment (xuzhang3) [inline website/docs/r/build_definition.html.markdown]: Document should point out if `repo_type` is `GitHubEnterprise`` ,   must use existing project and GitHub enterprise service connection.

**Review by xuzhang3 — CHANGES_REQUESTED** [review]
@Shubydo  Thanks for contribution and I left some comments on what you might need to change.

File: website/docs/r/build_definition.html.markdown (hunk 2 above)
Comment (shubydo) [reply website/docs/r/build_definition.html.markdown:14]: Team Foundation Server + Git. It's an example when `repo_type` is `TfsGit`

**Comment by shubydo** [comment]
> @Shubydo Thanks for contribution and I left some comments on what you might need to change.

Made the updates. Please let me know if there is anything else.

**Pull request merged**